File: Modul-04/SOAL 4/PRAK404-2310817210009-MuhammadIbnuSina.c
#include <stdio.h>

int main() {
    while (1) {

        printf("\nPilih Program\n");
        printf("1. Penjumlahan\n");
        printf("2. Pengurangan\n");
        printf("3. Perkalian\n");
        printf("4. Pembagian\n");
        printf("5. Exit\n");

        char PILIHAN;
        printf("Masukkan Pilihan : ");
        scanf("%d", &PILIHAN);

        float ANGKAPERTAMA, ANGKAKEDUA, HASIL;
        if (PILIHAN == 5) {
            printf("Terimakasih, telah menggunakan kalkulator MUHAMMAD IBNU SINA\n");
            break;
        } else if (PILIHAN == 1) {
            printf("Masukkan nilai pertama : ");
            scanf("%f", &ANGKAPERTAMA);
            printf("Masukkan nilai kedua : ");
            scanf("%f", &ANGKAKEDUA);
            HASIL = ANGKAPERTAMA + ANGKAKEDUA;
            printf("Hasil penjumlahan antara %.2f dengan %.2f adalah %.2f\n", ANGKAPERTAMA, ANGKAKEDUA, HASIL);
        } else if (PILIHAN == 2) {
            printf("Masukkan nilai pertama : ");
            scanf("%f", &ANGKAPERTAMA);
            printf("Masukkan nilai kedua : ");
            scanf("%f", &ANGKAKEDUA);
            HASIL = ANGKAPERTAMA - ANGKAKEDUA;
            printf("Hasil pengurangan antara %.2f dengan %.2f adalah %.2f\n", ANGKAPERTAMA, ANGKAKEDUA, HASIL);
        } else if (PILIHAN == 3) {
            printf("Masukkan nilai pertama : ");
            scanf("%f", &ANGKAPERTAMA);
            printf("Masukkan nilai kedua : ");
            scanf("%f", &ANGKAKEDUA);
            HASIL = ANGKAPERTAMA * ANGKAKEDUA;
            printf("Hasil pengkalian antara %.2f dengan %.2f adalah %.2f\n", ANGKAPERTAMA, ANGKAKEDUA, HASIL);
        } else if (PILIHAN == 4) {
            printf("Masukkan nilai pertama : ");
            scanf("%f", &ANGKAPERTAMA);
            printf("Masukkan nilai kedua : ");
            scanf("%f", &ANGKAKEDUA);
            HASIL = ANGKAPERTAMA / ANGKAKEDUA;
            printf("Hasil penmbagian antara %.2f dengan %.2f adalah %.2f\n", ANGKAPERTAMA, ANGKAKEDUA, HASIL);
        } else {
            printf("Input anda salah, silahkan coba lagi\n");
        }   
    }
    return 0;
}

File: Modul-04/SOAL 4/PRAK404-2310817210009-MuhammadIbnuSina.py
def kalkulator():
    while True:
        print("\nPilih program")
        print("1. Penjumlahan")
        print("2. Pengurangan")
        print("3. Perkalian")
        print("4. Pembagian")
        print("5. Exit")

        PILIHAN = input("Masukkan Pilihan : ")

        if PILIHAN == "5":
            print("Terimakasih, telah menggunakan kalkulator MUHAMMAD IBNU SINA")
            break

        if PILIHAN in ["1", "2", "3", "4"]:
            NILAIPERTAMA = float(input("Masukkan nilai pertama : "))
            NILAIKEDUA = float(input("Masukkan nilai kedua : "))

            if PILIHAN == "1":
                HASIL = NILAIPERTAMA + NILAIKEDUA
                print(f"Hasil Penjumlahan antara {NILAIPERTAMA:.2f} dengan {NILAIKEDUA:.2f} adalah {HASIL:.2f}")
            elif PILIHAN == "2":
                HASIL = NILAIPERTAMA - NILAIKEDUA
                print(f"Hasil Pengurangan antara {NILAIPERTAMA:.2f} dengan {NILAIKEDUA:.2f} adalah {HASIL:.2f}")
            elif PILIHAN == "3":
                HASIL = NILAIPERTAMA * NILAIKEDUA
                print(f"Hasil Perkalian antara {NILAIPERTAMA:.2f} dengan {NILAIKEDUA:.2f} adalah {HASIL:.2f}")
            elif PILIHAN == "4":
                if NILAIKEDUA == 0:
                    print("Pembagian dengan nol tidak diizinkan.")
                else:
                    HASIL = NILAIPERTAMA / NILAIKEDUA
                    print(f"Hasil Pembagian antara {NILAIPERTAMA:.2f} dengan {NILAIKEDUA:.2f} adalah {HASIL:.2f}")
        else:
            print("Input anda salah, silahkan coba lagi")

if __name__ == "__main__":
    kalkulator()